### PR TITLE
Gio library update

### PR DESCRIPTION
**Pull request by Sirmorrison**
Update gio library to the recent release.
* From the news release, Icon layout now receives the color parameter instead of the icon size.
* `op.Save()` has been deprecated
* clip.go `Add()` has been deprecated.. all instances to now use `push()`
* Remove custom scrollbar and float widget.

### CODE REVIEW

File: ui/decredmaterial/float.go [hunk 1]
```diff
@@ -2,98 +2,98 @@
...
-}
-
-// Changed reports whether the value has changed since
-// the last call to Changed.
-func (f *Float) Changed() bool {
-	changed := f.changed
-	f.changed = false
-	return changed
-}
+// import (
```
Comment (dreacot) [inline ui/decredmaterial/float.go]: why is this commented?

File: ui/decredmaterial/scrollbar.go [hunk 2]
```diff
@@ -2,163 +2,163 @@
...
-			if s.totalContentHeight <= totalVisibleHeight {
-				return layout.Dimensions{}
-			}
-
-			initialPosition := float32(s.container.Position.First) / float32(len(widgets))
-			return s.scrollbar.Layout(gtx, s.totalContentHeight, initialPosition)
-		}),
-	)
-}
+// import (
```
Comment (dreacot) [inline ui/decredmaterial/scrollbar.go]: why is this commented?

File: ui/page/proposal/proposals_page.go [hunk 3]
```diff
@@ -234,13 +234,11 @@ func (pg *ProposalsPage) initLayoutWidgets() {
 	pg.syncButton = new(widget.Clickable)
 
 	pg.infoIcon = pg.Icons.ActionInfo
-	pg.infoIcon.Color = pg.Theme.Color.Gray
 
 	pg.legendIcon = pg.Icons.ImageBrightness1
-	pg.legendIcon.Color = pg.Theme.Color.InactiveGray
+	// pg.legendIcon.Color = pg.Theme.Color.InactiveGray
```
Comment (dreacot) [inline ui/page/proposal/proposals_page.go]: why is this commented?

File: ui/page/proposal/proposal_details_page.go [hunk 4]
```diff
@@ -187,13 +188,14 @@ func (pg *proposalDetails) layoutInDiscussionState(gtx C) D {
 					})
 				}
 				icon := pg.successIcon
-				icon.Color = pg.Theme.Color.Primary
+				// todo icon size = values.MarginPadding20
```
Comment (dreacot) [inline ui/page/proposal/proposal_details_page.go]: this looks like it has been done on line 197

File: ui/page/send/send_confirm_modal.go [hunk 5]
```diff
@@ -156,9 +156,9 @@ func (scm *sendConfirmModal) Layout(gtx layout.Context) D {
 					return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
 						layout.Rigid(func(gtx C) D {
 							icon := scm.Icons.NavigationArrowForward
-							icon.Color = scm.Theme.Color.Gray3
 							return layout.Inset{Right: values.MarginPadding8}.Layout(gtx, func(gtx C) D {
-								return icon.Layout(gtx, values.MarginPadding15)
+								// todo:iconsize values.MarginPadding8
```
Comment (dreacot) [inline ui/page/send/send_confirm_modal.go]: if the todos for icon size are going to be done in a later PR then an issue should be opened so it's not forgotten

**Review by dreacot — CHANGES_REQUESTED** [review]
Overall the PR could do with a description to explain what has been done, because it doesn't seem to be just the gio library update

File: ui/decredmaterial/float.go (hunk 1 above)
Comment (Sirmorrison) [reply ui/decredmaterial/float.go]: The reason for the Library update was to remove this custom widget.

File: ui/decredmaterial/scrollbar.go (hunk 2 above)
Comment (Sirmorrison) [reply ui/decredmaterial/scrollbar.go]: The reason for the Library update was to remove this custom widget.

File: ui/decredmaterial/float.go (hunk 1 above)
Comment (Sirmorrison) [reply ui/decredmaterial/float.go]: This has been implemented on the gioui library

File: ui/decredmaterial/scrollbar.go (hunk 2 above)
Comment (Sirmorrison) [reply ui/decredmaterial/scrollbar.go]: This has been implemented on the gioui library

File: ui/page/proposal/proposals_page.go (hunk 3 above)
Comment (Sirmorrison) [reply ui/page/proposal/proposals_page.go]: The update does not support icon color struct variable.. so it is no longer needed.

File: ui/page/proposal/proposal_details_page.go (hunk 4 above)
Comment (Sirmorrison) [reply ui/page/proposal/proposal_details_page.go]: Removed... it no longer needed.

File: ui/page/send/send_confirm_modal.go (hunk 5 above)
Comment (Sirmorrison) [reply ui/page/send/send_confirm_modal.go]: This is resolved already.

File: ui/decredmaterial/votebar.go [hunk 6]
```diff
@@ -233,12 +231,13 @@ func (v *VoteBar) LayoutWithLegend(gtx C) D {
 	)
 }
 
-func (v *VoteBar) layoutIconAndText(gtx C, lbl Label, count float32) D {
+func (v *VoteBar) layoutIconAndText(gtx C, lbl Label, count float32, clr color.NRGBA) D {
```
Comment (dreacot) [inline ui/decredmaterial/votebar.go:234]:
```suggestion
func (v *VoteBar) layoutIconAndText(gtx C, label Label, count float32, color color.NRGBA) D {
```

I don't think it's necessary to shorten words that aren't extremely long, as it reduces code readability

There are several cases of this in the code, so i guess we can proceed with this and do a cleanup later

**Review by dreacot — APPROVED** [review]

File: ui/decredmaterial/votebar.go (hunk 6 above)
Comment (Sirmorrison) [reply ui/decredmaterial/votebar.go:234]: This method and file has been reviewed in Pr https://github.com/planetdecred/godcr/pull/638. Also, more cleanup will be carried out when the new UI is ready.

**Pull request merged**